### PR TITLE
CDK itests refactoring

### PR DESCRIPTION
**Pull request by odockal**
- added 120 second treshold for each server adapter start operation - CDK - 216 (resolved)
	- finished adding of mocking files
	- discovery itests repaired
	- discovery itests refactored to run faster at the expense of test separation
	- removed deprecated tests - cdk 2.x itests removed from suites
	- added wait condition to accept dfferent problem dialogs during cdk start up
	- added skipping of cdk rhel image registration
	- added test case with verification of cdk rhel image registration

Signed-off-by: Ondrej Dockal <odockal@redhat.com>

# Pull Request Checklist
## General
* Is this a blocking issue or new feature? If yes, QE needs to +1 this PR

## Code
* Are method-/class-/variable-names meaningful?
* Are methods concise, not too long?
* Are catch blocks catching precise Exceptions only (no catch all)?

## Testing
* Are there unit-tests?
* Are there integration tests (or at least a jira to tackle these)?
* Is the non-happy path working, too?
* Are other parts that use the same component still working fine?

## Function
* Does it work?

### CODE REVIEW

File: test-framework/org.jboss.tools.cdk.reddeer/src/org/jboss/tools/cdk/reddeer/server/ui/CDEServer.java [hunk 1]
```diff
@@ -69,59 +82,113 @@ protected void operateServerState(String menuItem, final ServerState resultState
...
+
+		@Override
+		public boolean test() {
+			if (expectDialog) {
+				if (sslDialog.test()) {
+					confirmSSLCertificateDialog();
+					return true;
+				}
+			}
+			for (String dialogName : problemDialogs.keySet()) {
```
[the remark this answers is not in the document]
Comment (adietish) [reply test-framework/org.jboss.tools.cdk.reddeer/src/org/jboss/tools/cdk/reddeer/server/ui/CDEServer.java]: seems legit too, would spare a get for an iterated key.

File: test-framework/org.jboss.tools.cdk.reddeer/src/org/jboss/tools/cdk/reddeer/server/ui/CDEServer.java [hunk 2]
```diff
@@ -39,6 +44,10 @@
```
[the remark this answers is not in the document]
Comment (odockal) [reply test-framework/org.jboss.tools.cdk.reddeer/src/org/jboss/tools/cdk/reddeer/server/ui/CDEServer.java]: I was naming the class base on the fact that adapter itself is called 'Container Development Environemnt'... no problem to change the name to be aligned with test classes.

**Comment by odockal** [comment]
@adietish All remarks processed and pushed. thanks Andre!

**Comment by odockal** [comment]
@adietish @jkopriva Branch is unfrozen, would be great to have this merged unless you object anything.

**Comment by jkopriva** [comment]
@odockal LGTM, can you also post link to green build?

**Comment by jkopriva** [comment]
testPR

**Comment by odockal** [comment]
@adietish Do you mind taking care of this? We got compare baseline check failure, we need to bump up the plugins' versions.

**Comment by adietish** [comment]
@odockal any change that we'd bump the plugin version so that we get a green build here?

**Comment by odockal** [comment]
@adietish I believe that we need to bump up everything in openshift/plugins, features, other projects etc. would be better to have another PR for this and after merging I would rebase, WDYT? Let ask @nickboldt when he will have time for this (and maybe even some automation script as well).

**Comment by odockal** [comment]
testPR

**Comment by adietish** [comment]
testPR

**Pull request merged**